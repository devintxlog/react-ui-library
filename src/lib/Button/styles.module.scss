.button {
  outline: 0;
  border: 0;
  cursor: pointer;
}

.link {
  text-decoration: none;
  color: var(--colorPrimaryBase);
  transition: color 0.3s;

  &:hover {
    color: var(--colorPrimaryLight);
  }

  &:active {
    color: var(--colorPrimaryLight);
  }
}

.plain {
  background-color: transparent;
}

.primary {
  color: white;
  border-radius: 3px;
  transition: background-color 0.3s;
  background-color: var(--colorPrimaryBase);

  &:hover {
    background-color: var(--colorPrimaryLight);
  }

  &:active {
    background-color: var(--colorPrimaryLight);
  }
}

.small {
  font-size: 12px;
  line-height: 12px;
  padding: 6px 10px;
}

.medium {
  font-size: 16px;
  line-height: 16px;
  padding: 10px 15px;
}

.large {
  font-size: 20px;
  line-height: 20px;
  padding: 10px 20px;
}