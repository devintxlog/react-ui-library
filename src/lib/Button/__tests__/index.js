import React from 'react'

import { shallow } from 'enzyme'
import Button from '..'

describe('Button', () => {
  //define mock functions
  const onClickMock = jest.fn()

  //define default props
  let props = {
    onClick: onClickMock
  }

  //create a shallow wrapper than can be reused
  let wrapper
  //initialize the components
  beforeEach(() => {
    wrapper = shallow(<Button {...props}/>)
  })

  afterEach(() => {
    onClickMock.mockReset()
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  }),
  describe('when fullWidth prop is true', () =>{
    it('renders correctly', () => {
      wrapper.setProps({ 
        fullWidth: true
      })
      expect(wrapper).toMatchSnapshot()
    })
  }),
  describe('when text prop is Example Text', () =>{
    it('renders correctly', () => {
      wrapper.setProps({ 
        text: `Example Text`
      })
      expect(wrapper).toMatchSnapshot()
    })
  }),
  describe('when fontSize prop is 14', () =>{
    it('renders correctly', () => {
      wrapper.setProps({ 
        fontSize: 14
      })
      expect(wrapper).toMatchSnapshot()
    })
  }),
  describe('when type prop', () =>{
    describe('is primary', () =>{
      it('renders correctly', () => {
        wrapper.setProps({ 
          type: `primary`
        })
        expect(wrapper).toMatchSnapshot()
      })
    }),
    describe('is plain', () =>{
      it('renders correctly', () => {
        wrapper.setProps({ 
          type: `plain`
        })
        expect(wrapper).toMatchSnapshot()
      })
    }),
    describe('is link', () =>{
      it('renders correctly', () => {
        wrapper.setProps({ 
          type: `link`,
          linkTo: '/'
        })
        expect(wrapper).toMatchSnapshot()
      })
    }),
    describe('is routedLink', () =>{
      it('renders correctly', () => {
        wrapper.setProps({ 
          type: `routedLink`,
          linkTo: '/'
        })
        expect(wrapper).toMatchSnapshot()
      })
    })
  }),
  describe('when handleOnClick method is called', () => {
    it('calls the correct function', () => {
      expect(wrapper.instance().handleOnClick()).toEqual(undefined)
      expect(onClickMock).toHaveBeenCalled()
    })
  }),
  describe('when user clicks', () => {
    it('calls the correct function', () => {
      const buttonElement = wrapper.find('.button')
      buttonElement.simulate('click')
      expect(onClickMock).toHaveBeenCalledTimes(1)
      expect(onClickMock).toHaveBeenCalledWith()
    })
  })
})