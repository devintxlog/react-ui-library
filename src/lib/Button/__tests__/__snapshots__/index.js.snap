// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="primary"
  onClick={[MockFunction]}
  type="button"
>
  None Provided
</button>
`;

exports[`Button when fontSize prop is 14 renders correctly 1`] = `
<button
  className="primary"
  onClick={[MockFunction]}
  type="button"
>
  None Provided
</button>
`;

exports[`Button when fullWidth prop is true renders correctly 1`] = `
<button
  className="primary fullWidth"
  onClick={[MockFunction]}
  type="button"
>
  None Provided
</button>
`;

exports[`Button when submitsForm prop is true renders correctly 1`] = `
<button
  className="primary"
  onClick={[MockFunction]}
  type="submit"
>
  None Provided
</button>
`;

exports[`Button when text prop is Example Text renders correctly 1`] = `
<button
  className="primary"
  onClick={[MockFunction]}
  type="button"
>
  Example Text
</button>
`;

exports[`Button when type prop is link renders correctly 1`] = `
<a
  className="primary"
  href="/"
  rel="noopener noreferrer"
  target="_blank"
>
  None Provided
</a>
`;

exports[`Button when type prop is plain renders correctly 1`] = `
<button
  className="plain"
  onClick={[MockFunction]}
  type="button"
>
  None Provided
</button>
`;

exports[`Button when type prop is primary renders correctly 1`] = `
<button
  className="primary"
  onClick={[MockFunction]}
  type="button"
>
  None Provided
</button>
`;

exports[`Button when type prop is routedLink renders correctly 1`] = `
<button
  className="primary"
  onClick={[MockFunction]}
  type="button"
>
  None Provided
</button>
`;
